### ISSUE TITLE
Adding new an OEC member removes other members from member list

### ISSUE THREAD

**Issue by acka47** [hbz/oerworldmap#469]
Merging duplicates (see #440) I deleted three entries of OER members adding the `memberOf` relation to the existing entries (see e.g. https://github.com/hbz/oerworldmap/issues/440#issuecomment-158413800). Obviously, this resulted in updating the OEC entry (https://oerworldmap.org/resource/urn:uuid:ff56c436-7e76-11e5-b76b-54ee7558c81f) removing the `member` information of all other OEC members. The `memberOf` data of these OEC members remained intact, though, as did the embedded info about the OEC see e.g. https://oerworldmap.org/resource/urn:uuid:706ad082-77eb-11e5-9f9f-c48e8ff4ea31 (snippet):

```javascript
  "@id" : "urn:uuid:706ad082-77eb-11e5-9f9f-c48e8ff4ea31",
  "memberOf" : [ {
    "image" : "http://www.oeconsortium.org/wp-content/themes/ocwc-mainsite/images/logo.png",
    "@type" : "Organization",
    "name" : [ {
      "@value" : "Open Education Consortium",
      "@language" : "en"
    } ],
    "member" : [ {
      "@type" : "Organization",
      "name" : [ {
        "@value" : "University of Southern Queensland",
        "@language" : "en"
      } ],
      "@id" : "urn:uuid:706b65ce-77eb-11e5-9f9f-c48e8ff4ea31"
    }, {
      "@type" : "Organization",
      "name" : [ {
        "@value" : "Lektorium LLC.",
        "@language" : "en"
      } ],
      "@id" : "urn:uuid:7069671a-77eb-11e5-9f9f-c48e8ff4ea31"
    }, 
    ...
```

**Comment by acka47** [comment]
We couldn't reconstruct this issue. Usually, adding and/or removing an OEC member results in the desired behaviour.

I am not sure, though, whether I did all the edits via the web forms. I will look into the history of my home office computer tomorrow and clarify this.

**Comment by acka47** [comment]
Actually, I did one POST via curl:

`curl -i -H "Content-Type: application/json" -X POST -d @cccoer.json -u oerworldmap-9005 https://oerworldmap.org/resource/`

After this, I deleted the record with `curl -X DELETE https://oerworldmap.org/resource/urn:uuid:706904c8-77eb-11e5-9f9f-c48e8ff4ea31 -u oerworldmap-9005`.

The deletion of the `member` triples at OEC may have happened by then. I created a [gist with cccoer.json](https://gist.github.com/acka47/507b3e04288e4b9a4410).

**BUT**, after doing this, I added the `memberOf` statement to the [Anne Arundel Community College entry](https://oerworldmap.org/resource/urn:uuid:de67e70c-3b7b-11e5-8d59-54ee7558c81f) and deleted the duplicate. And you won't find this college as member of [OEC](https://oerworldmap.org/resource/urn:uuid:ff56c436-7e76-11e5-b76b-54ee7558c81f). So there must also have been a problem after doing the POST via curl.

**Comment by acka47** [comment]
Here are the relevant parts of my bash history for reconstructing what I did:

1. Testing on beta. Obviously, I modified the [CCCOER entry](https://beta.oerworldmap.org/resource/urn:uuid:2adb2bb2-3073-11e5-b60d-bfcc6080a842) via the web form and then deleted the duplicate: `curl -X DELETE https://beta.oerworldmap.org/resource/urn:uuid:706904c8-77eb-11e5-9f9f-c48e8ff4ea31 -u oerworldmap-9004`

2. I then GET the correct entry from beta and store it locally: `getjson https://beta.oerworldmap.org/resource/urn:uuid:2adb2bb2-3073-11e5-b60d-bfcc6080a842 > cccoer.json`

3. I post this entry to production: `curl -i -H "Content-Type: application/json" -X POST -d @cccoer.json -u oerworldmap-9005 https://oerworldmap.org/resource/` and also deleted the duplicate on production: `curl -X DELETE https://oerworldmap.org/resource/urn:uuid:706904c8-77eb-11e5-9f9f-c48e8ff4ea31 -u oerworldmap-9005`

4. Then I modified the other entries in the web form, deleting duplicates afterwards:
  - `curl -X DELETE https://oerworldmap.org/resource/urn:uuid:7069aa18-77eb-11e5-9f9f-c48e8ff4ea31 -u oerworldmap-9005`
  - `curl -X DELETE https://oerworldmap.org/resource/urn:uuid:706976c4-77eb-11e5-9f9f-c48e8ff4ea31 -u oerworldmap-9005`
  - `curl -X DELETE https://oerworldmap.org/resource/urn:uuid:7069464a-77eb-11e5-9f9f-c48e8ff4ea31 -u oerworldmap-9005`
  - `curl -X DELETE https://oerworldmap.org/resource/urn:uuid:706a9f4a-77eb-11e5-9f9f-c48e8ff4ea31 -u oerworldmap-9005`

**Comment by philboeselager** [comment]
There is something about the member's IDs: as you can see in the [cccoer.json](https://gist.githubusercontent.com/acka47/507b3e04288e4b9a4410/raw/73e5de2e5215a9b86cf2891b9e6832dcfc439138/cccoer.json), the member's IDs are the same as used in deleted duplicate, that is

    urn:uuid:706976c4-77eb-11e5-9f9f-c48e8ff4ea31

though the new container resource has the ID

    urn:uuid:2adb2bb2-3073-11e5-b60d-bfcc6080a842

As soon as the duplicate has been deleted, these member's IDs point to nothing and therefore the members disappear. Does this make sense?

**Comment by acka47** [comment]
I don't see the problem. When I posted [cccoer.json](https://gist.githubusercontent.com/acka47/507b3e04288e4b9a4410/raw/73e5de2e5215a9b86cf2891b9e6832dcfc439138/cccoer.json), the duplicates still were there as OEC. I deleted them afterwards. Thus, it shouldn't be a problem.

Also, the CCCoer ID is the same for the container and as member of the OEC: `urn:uuid:2adb2bb2-3073-11e5-b60d-bfcc6080a842`

**Comment by philboeselager** [comment]
Cannot reproduce. Therefore, agreed to leave it as is. Except for: sent a repaired file "Open Education Consortium" to @acka47 , containing all new members, as well as the old ones without the ones to be deleted mentioned above.
@acka47 : please check this file, POST to oerworldmap.org (if appears to be good) and close this ticket.

**Comment by acka47** [comment]
Puh, I would have preferred to do this the way I suggested yesterday when we talked. I had to do some more work on the file @philboeselager sent me. Only one or two of the five entries above was deleted from the file, I removed the other three resulting in [this file](https://gist.github.com/acka47/e0232d623d3bd15c06db), added urn:uuid:de67e70c-3b7b-11e5-8d59-54ee7558c81f (which was missing) and tried posting the file to oerworldmap. It wouldn't work:

```
$ curl -v -s -H "Content-Type: application/json" -X POST -d @mixed_consortium3.json -u oerworldmap-9005 https://oerworldmap.org/resource/
Enter host password for user 'oerworldmap-9005':
* Hostname was NOT found in DNS cache
*   Trying 193.30.112.94...
* Connected to oerworldmap.org (193.30.112.94) port 443 (#0)
* successfully set certificate verify locations:
*   CAfile: none
  CApath: /etc/ssl/certs
* SSLv3, TLS handshake, Client hello (1):
* SSLv3, TLS handshake, Server hello (2):
* SSLv3, TLS handshake, CERT (11):
* SSLv3, TLS handshake, Server key exchange (12):
* SSLv3, TLS handshake, Server finished (14):
* SSLv3, TLS handshake, Client key exchange (16):
* SSLv3, TLS change cipher, Client hello (1):
* SSLv3, TLS handshake, Finished (20):
* SSLv3, TLS change cipher, Client hello (1):
* SSLv3, TLS handshake, Finished (20):
* SSL connection using ECDHE-RSA-AES256-GCM-SHA384
* Server certificate:
* 	 subject: C=DE; ST=Nordrhein-Westfalen; L=Koeln; O=Hochschulbibliothekszentrum NRW; OU=Rechenzentrum; CN=oerworldmap.org
* 	 start date: 2015-03-05 12:09:11 GMT
* 	 expire date: 2019-07-09 23:59:00 GMT
* 	 subjectAltName: oerworldmap.org matched
* 	 issuer: C=DE; O=Hochschulbibliothekszentrum NRW; OU=Rechenzentrum; CN=hbz NRW CA - G02; emailAddress=pki@hbz-nrw.de
* 	 SSL certificate verify ok.
* Server auth using Basic with user 'oerworldmap-9005'
> POST /resource/ HTTP/1.1
> Authorization: Basic b2Vyd29ybGRtYXAtOTAwNTpvZXJ3bTIwMTU=
> User-Agent: curl/7.35.0
> Host: oerworldmap.org
> Accept: */*
> Content-Type: application/json
> Content-Length: 468851
> Expect: 100-continue
> 
< HTTP/1.1 100 Continue
< HTTP/1.1 404 Not Found
< Date: Fri, 04 Dec 2015 09:11:25 GMT
* Server Apache/2.4.10 (Linux/SUSE) is not blacklisted
< Server: Apache/2.4.10 (Linux/SUSE)
< Content-Type: text/html; charset=utf-8
< Transfer-Encoding: chunked
* HTTP error before end of send, stop sending
< 


<!DOCTYPE html>
<html>
	<head>
		<title>Action not found</title>
		<link rel="shortcut icon" href="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAYAAAAf8/9hAAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAlFJREFUeNqUU8tOFEEUPVVdNV3dPe8xYRBnjGhmBgKjKzCIiQvBoIaNbly5Z+PSv3Aj7DSiP2B0rwkLGVdGgxITSCRIJGSMEQWZR3eVt5sEFBgTb/dN1yvnnHtPNTPG4PqdHgCMXnPRSZrpSuH8vUJu4DE4rYHDGAZDX62BZttHqTiIayM3gGiXQsgYLEvATaqxU+dy1U13YXapXptpNHY8iwn8KyIAzm1KBdtRZWErpI5lEWTXp5Z/vHpZ3/wyKKwYGGOdAYwR0EZwoezTYApBEIObyELl/aE1/83cp40Pt5mxqCKrE4Ck+mVWKKcI5tA8BLEhRBKJLjez6a7MLq7XZtp+yyOawwCBtkiBVZDKzRk4NN7NQBMYPHiZDFhXY+p9ff7F961vVcnl4R5I2ykJ5XFN7Ab7Gc61VoipNBKF+PDyztu5lfrSLT/wIwCxq0CAGtXHZTzqR2jtwQiXONma6hHpj9sLT7YaPxfTXuZdBGA02Wi7FS48YiTfj+i2NhqtdhP5RC8mh2/Op7y0v6eAcWVLFT8D7kWX5S9mepp+C450MV6aWL1cGnvkxbwHtLW2B9AOkLeUd9KEDuh9fl/7CEj7YH5g+3r/lWfF9In7tPz6T4IIwBJOr1SJyIGQMZQbsh5P9uBq5VJtqHh2mo49pdw5WFoEwKWqWHacaWOjQXWGcifKo6vj5RGS6zykI587XeUIQDqJSmAp+lE4qt19W5P9o8+Lma5DcjsC8JiT607lMVkdqQ0Vyh3lHhmh52tfNy78ajXv0rgYzv8nfwswANuk+7sD/Q0aAAAAAElFTkSuQmCC">
	    <style>
		    html, body, pre {
		        margin: 0;
		        padding: 0;
		        font-family: Monaco, 'Lucida Console', monospace;
		        background: #ECECEC;
		    }
		    h1 {
		        margin: 0;
		        background: #AD632A;
		        padding: 20px 45px;
		        color: #fff;
		        text-shadow: 1px 1px 1px rgba(0,0,0,.3);
		        border-bottom: 1px solid #9F5805;
		        font-size: 28px;
		    }
		    p#detail {
		        margin: 0;
		        padding: 15px 45px;
		        background: #F6A960;
		        border-top: 4px solid #D29052;
		        color: #733512;
		        text-shadow: 1px 1px 1px rgba(255,255,255,.3);
		        font-size: 14px;
		        border-bottom: 1px solid #BA7F5B;
		    }
		</style>
	</head>
	<body>
		<h1>Action not found</h1>

		<p id="detail">
			For request 'GET /error/HTTP_BAD_GATEWAY.html.var'
		</p>

	</body>
</html>







* Closing connection 0
* SSLv3, TLS alert, Client hello (1):

```

**Comment by acka47** [comment]
Just talked wth @literarymachine about this on the phone. I will create an OEC record just with `member` and a list of member URIs but won't post it before Felix tells me so. Felix will look into the error I posted above.

**Comment by acka47** [comment]
By now, the OEC entry is fixed, see https://oerworldmap.org/resource/urn:uuid:ff56c436-7e76-11e5-b76b-54ee7558c81f. Closing.

**Comment by acka47** [comment]
Again, there are a lot of members missing in the OEC entry: https://oerworldmap.org/resource/urn:uuid:ff56c436-7e76-11e5-b76b-54ee7558c81f Obviously, this isn't fixed. Reopening.

**Comment by acka47** [comment]
We should not try to fix this in the backend as the underlying problem will be fixed with #528. We will repair the OEC entry once again manually though.

**Comment by acka47** [comment]
@literarymachine Please assign this to me if you don't manage to fix this today.

**Comment by literarymachine** [comment]
See script in etherpad. Current fixed file sent by mail.

**Comment by acka47** [comment]
For now, the OEC entry is ok. I will take a look at it once and again and run the script if I encounter any problems. The issue will be closed when #528 is finished.

**Comment by acka47** [comment]
Unassigning myself. Depending on how #528 will take, maybe @literarymachine will take a look at https://github.com/orgs/hbz/teams/owners once in a while. Otherwisejust close it when the triple store backend is in production.
